impl/src/main/java/org/jboss/weld/bean/proxy/InterceptedSubclassFactory.java

```diff
@@ -238,6 +254,16 @@ void doReturn(CodeAttribute b, ClassMethod method) {
         }
     }
 
+    @Override
+    public Set<Class<?>> getAdditionalInterfaces() {
```

Comment:
I think we should not override the `getAdditionalInterfaces()` method but use the `interfacesToInspect` directly where needed. We don't want to extend the `additionalInterfaces` set as defined in `ProxyFactory` after all.